### ISSUE TITLE
Allow option to specify image pull policy for component container image

### ISSUE THREAD

**Issue by awisniew90** [redhat-developer/odo#4494]
/kind feature

## Which functionality do you think we should add?

Odo should allow users to specify an image pull policy for a component's container image.  

## Why is this needed?

Using local images for components is particularly necessary when testing out stack image changes. Without being able to change the image pull policy, a local image cannot be used.

**Comment by awisniew90** [comment]
Related to: https://github.com/devfile/api/issues/89

**Comment by dharmit** [comment]
/remove-lifecycle rotten
/area devfile
/lifecycle frozen

Since there's already an issue about this on devfile side, I'm setting `lifecycle/frozen` to this issue. It first needs to be addressed in devfile api and library, IMO.

**Comment by kadel** [comment]
recently released odo v3.6 added support for `pod-overrides` and `container-overrides`. This devfile features allows specifying imagePullPolicy.

Demo can be found in release notes https://odo.dev/blog/odo-v3.6.0/#extend-pod-and-container-specification-with-devfile-attributes-pod-overrides-and-container-overrides

adding the following lines to `container` component will change imagePullPolicy

```yaml
    attributes:
      container-overrides:
        imagePullPolicy: IfNotPresent
```

/close
